OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -154,7 +154,7 @@ public override bool Tick(Actor self)
 
 			// Move into range of the target.
 			if (!target.IsInRange(pos, lastVisibleMaximumRange) || target.IsInRange(pos, minimumRange))
-				QueueChild(aircraft.MoveWithinRange(target, minimumRange, lastVisibleMaximumRange, target.CenterPosition, Color.Red));
+				QueueChild(aircraft.MoveWithinRange(target, minimumRange, lastVisibleMaximumRange, checkTarget.CenterPosition, Color.Red));
```

Comment:
So should we be checking for `checkTarget.IsInRange...` in the condition above? It seems odd to check whether we're in range of one thing, and then move towards a different thing.

(If the condition should change, then there might be other changes since we're already doing some such checks earlier on, but to start with just understanding if the condition should be altered too would be good)

Reply:
I see you adjusted the code.

As written now, this seems odd as on line 144 above we have
```c#
if (checkTarget.IsInRange(pos, lastVisibleMaximumRange) && !checkTarget.IsInRange(pos, minimumRange) && useLastVisibleTarget)
    return true;
```

which is bailing out if it is specifically having to use the last target. If later on we're moving into range of `checkTarget` per this PR's suggestion, then that feels.... weird? Why can we bail early in the `useLastVisibleTarget = true` case, but need to keep going for the `useLastVisibleTarget = false` case?

It might be that this logic is all fine and I just need an explanation as to why this all makes sense in both the `useLastVisibleTarget = true` and `useLastVisibleTarget = false` scenarios - or maybe there is something off about this that I can't quite put into words. If you can do anything to clear up my confusion, that would be very helpful.

Reply:
The duplicate range check and move order was always present ([see FlyAttack before the last relevant commit](https://github.com/OpenRA/OpenRA/blob/06cc47847d8c6a921bc105d79b4e0a1f8d7fc1b3/OpenRA.Mods.Common/Activities/Air/FlyAttack.cs#L147)) and I aimed to preserve the existing order of checks. The check for `useLastVisibleTarget` could be moved to the final if block, I could experiment to see if the now unskipped takeoff order causes any regressions

Reply:
Updated